src/ripple/app/main/Application.cpp

```diff
@@ -1344,26 +1344,34 @@ ApplicationImp::setup()
 
     // Configure the amendments the server supports
     {
-        auto const& sa = detail::supportedAmendments();
-        std::vector<std::string> saHashes;
-        saHashes.reserve(sa.size());
-        for (auto const& name : sa)
-        {
-            auto const f = getRegisteredFeature(name);
-            BOOST_ASSERT(f);
-            if (f)
-                saHashes.push_back(to_string(*f) + " " + name);
-        }
-        Section supportedAmendments("Supported Amendments");
-        supportedAmendments.append(saHashes);
+        auto buildAmendmentList =
+            [](Section section, std::vector<std::string> const& amendments) {
+                std::vector<std::string> hashes;
+                hashes.reserve(amendments.size());
+                for (auto const& name : amendments)
+                {
+                    auto const f = getRegisteredFeature(name);
+                    assert(f);
```

Comment:
Just curious: How do we decide when to use `BOOST_ASSERT` vs `assert`? I guess they are synonymous?

Reply:
They're basically synonymous, but the boost version adds the dependency on boost. In general, AFAIK, we're moving away from `BOOST_ASSERT`, but nobody has bothered to do a global search and replace.